src/object/srv_ec_aggregate.c

```diff
@@ -2212,6 +2212,34 @@ agg_obj_is_leader(struct ds_pool *pool, struct daos_oclass_attr *oca,
 	return rc;
 }
 
+static bool
+agg_filter(vos_iter_desc_t *desc, void *cb_arg)
+{
+	struct ec_agg_param	*agg_param = (struct ec_agg_param *)cb_arg;
+	struct ec_agg_pool_info	*info = &agg_param->ap_pool_info;
+	struct daos_oclass_attr  oca;
+	int			 rc;
+
+	if (desc->id_type != VOS_ITER_OBJ)
+		return false;
+
+	rc = dsc_obj_id2oc_attr(desc->id_oid.id_pub, &info->api_props, &oca);
+	if (rc) {
+		D_ERROR("Skip object("DF_OID") with unknown class(%u)\n",
+			DP_OID(desc->id_oid.id_pub),
+			daos_obj_id2class(desc->id_oid.id_pub));
+		return true;
+	}
+
+	if (!daos_oclass_is_ec(&oca)) { /* Skip non-EC object */
+		D_DEBUG(DB_EPC, "Skip oid:"DF_UOID" non-ec obj\n",
+			DP_UOID(desc->id_oid));
+		return true;
+	}
```

Comment:
I thought we need check credit here as well? like agg_iterate_pre_cb()

Reply:
I'll have to think on that...it's a more complicated place to yield.   What we could do easily is increment the credit

Reply:
Without yield in pre_check callback, we can only optimize the EC iteration a bit (to avoid ilog check), but EC iteration will still to huge amount of busy loops.
I'm wondering if we need to make such an optimizing to 2.02?

Reply:
I'll give it some thought tomorrow.   Maybe I can just do something like acts in the iterator and be a little more generic